doc/ug/dv_methodology/index.md

```diff
@@ -299,61 +299,116 @@ Once the regression tool is developed, it will provide a way to enable these cap
+This observes events or series of events.
+As an example think of the TL UL register bus used in OpenTitan.
+A cover property cover point could be the handshaking between valid and ready.
+SVA also allows the Design engineer to add cover procedures and variables not visible on output pins.
+Note, an assertion precondition counts as a cover point.
+
+
+#### Do we need both types of coverage
+
+Reaching a 100% code coverage does not tell you the whole story.
+Think of the simple example of an AND gate with inputs A, B, and output O.
+To get to 100% code coverage only two different input combinations are needed.
+00 and 11, these two will produce all possible outcomes of O.
+
+![single AND gate](dv_method_single_gate.svg)
+
+
+The coverage will indicate that all code was exercised.
+But we do not know that our design works as intended.
+All we know is that A, B, and O have been observed to take on both logic 0 and 1.
+We could not say for certain that the design was in fact an AND gate: it could just as easily be an OR gate.
+So we need functional coverage to tell us this.
+The first thing functional coverage will tell us is if we observed all possible values on the inputs.
+And by adding a cross between the inputs and the output it will tell us which gate we are looking at.
+
+
+Reaching 100% functional coverage is not enough either.
+Remember functional coverage requires the designer to manually add coverage point into the design.
+Going back to our AND gate, let us say we take two of these and OR the outputs of the two.
+
+![multi gate](dv_method_multi_gate.svg)
+
 
-Code coverage can reach 100% but it may not be sufficient to indicate whether all interesting combinations of scenarios were exercised (such as boundary conditions, concurrent scenarios and different CSR fields holding specific combinations of values resulting in the DUT being exercised in interesting ways).
-Hence, developing a detailed functional coverage model in conjunction with constrained random test sequences is key to high quality DV effort.
+If we only add the cover points from our original example, these will still exercise the new output of our system and therefore result in reaching 100% functional coverage, but half of the design was not exercised.
+This is called a coverage hole and code coverage would have indicated that a part of the code was never exercised.
 
-Functional coverage can also reach 100% while the code coverage may not.
-This would indicate that the testplan is either incomplete or there are undocumented or unreachable features in the design.
-As such they can be addressed accordingly through either updating the design specification and augmenting the testplan / written tests, or by optimizing the design to remove unreachable logic if possible.
+While functional coverage will tell you if your design is working correctly, code coverage will highlight if your testplan is incomplete, and if there are any undcovered/unreachable features in the design.
```

Comment:
"undcovered". Also, I think this should be "or", not "and" (it spots if either thing happens, not only if both have happened).

Reply:
fixed

Reply:
did you make the same spelling mistake as me on purpose here :D